quickwit-storage/src/storage.rs

```diff
@@ -83,8 +83,8 @@ impl From<&'static [u8]> for PutPayload {
 /// object storage treat them. This means when directory separators a present
 /// in the storage operation path, the storage implementation should create and remove transparently
 /// these intermediate directories.
-#[cfg_attr(any(test, feature = "testsuite"), mockall::automock)]
 #[async_trait]
+#[cfg_attr(any(test, feature = "testsuite"), mockall::automock)]
```

Comment:
any reason for this change?

Reply:
Yes, so placing the `mockall::` after make the async methods visible to the Mock, without this change, the async function are just generated in mockall as normal functions. I needed this for testing concurrency on the storage. this test has been scrapped because it's not needed, but I kept this as it convey what we do here.

Reply:
I remember you mentionned that in the chat :)